### ISSUE TITLE
Improve --republish-buried

### ISSUE THREAD

**Issue by smacker** [src-d/borges#209]
Currently, it reads everything from `borges.buriedQueue` in memory and only then start publishing.
With 116959 messages in `borges.buriedQueue` it takes ages to return them.

**Comment by smacker** [comment]
run locally for 95698 messages. And it was so slow that I gave up.
Then I tried with 1000 messages
```
$ time go run cli/borges/*.go producer --republish-buried
here rovers
INFO[01-24|23:38:11] starting up                              module=borges mode=producer caller=producer.go:49
EROR[01-24|23:38:11] error obtaining next job                 module=borges mode=producer error="dial tcp 0.0.0.0:5432: getsockopt: connection refused" caller=producer.go:72
^Csignal: interrupt

real	1m4.605s
user	0m2.595s
sys	0m0.843s
```
(I pressed maybe after a second after it finished)

So here is 3 bug:
1) slow everything in once instead of 1 by 1
2) error obtaining next job
3)  very very nasty. `RepublishBuried` isn't atomic. So if I run command, don't wait it to finish and stop by `ctrl-c`, some messages get lost. That's why there only 95698 instead of 100000 messages.

**Comment by ajnavarro** [comment]
`RepublishBuried` must be atomic *per message*. Can you check if you already have messages in buried queue? the number should be 100000-95698=4302

**Comment by smacker** [comment]
Yes. It's not atomic at all. Messages just get lost. It gets better with my PR. In the worst case, it will lose 1 message now.
I'm going to try to improve it further later.
